io/nvmf/nvmftest.py

```diff
@@ -0,0 +1,180 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+"""
+NVM Express over Fabrics defines a common architecture that supports a
+range of storage networking fabrics for NVMe block storage protocol over
+a storage networking fabric.
+"""
+
+import os
+import json
+from avocado import Test
+from avocado import main
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process, linux_modules, genio
+import yaml
+
+
+class NVMfTest(Test):
+    """
+    NVM-Express over Fabrics using nvme-cli.
+    """
+
+    def setUp(self):
+        """
+        Sets up NVMf configuration
+        """
+        self.nvme_ns = self.params.get('namespace', default='')
+        self.peer_ip = self.params.get('peer_ip', default='')
+        if not self.nvme_ns:
+            self.cancel("No namespace provided")
+        if not self.peer_ip:
+            self.cancel("No peer_ip provided")
+        smm = SoftwareManager()
+        if not smm.check_installed("nvme-cli") and not \
+                smm.install("nvme-cli"):
+            self.cancel('nvme-cli is needed for the test to be run')
+        if not linux_modules.module_is_loaded("nvme-rdma"):
+            linux_modules.load_module("nvme-rdma")
+        self.cfg_tmpl = os.path.join(self.datadir, "nvmf_template.cfg")
+        self.cfg_file = os.path.join(self.datadir, "nvmf.cfg")
+        self.nvmf_discovery_file = "/etc/nvme/discovery.conf"
+
+    def create_cfg_file(self):
+        """
+        Creates the config file for nvmetcli to use in the target
+        """
+        with open(self.cfg_tmpl, "r") as cfg_fp:
+            cfg = yaml.safe_load(cfg_fp)
+
+        cfg["subsystems"][0]["namespaces"][0]["device"]["nguid"] = "1".zfill(
+            32)
+        cfg["subsystems"][0]["namespaces"][0]["device"]["path"] = self.nvme_ns
+        cfg["subsystems"][0]["namespaces"][0]["nsid"] = "1"
+        cfg["subsystems"][0]["nqn"] = "mysubsys1"
+        cfg["ports"][0]["addr"]["traddr"] = self.peer_ip
+        cfg["ports"][0]["subsystems"][0] = "mysubsys1"
+        cfg["ports"][0]["portid"] = "1"
+
+        with open(self.cfg_file, "w") as cfg_fp:
+            json.dump(cfg, cfg_fp, indent=2)
+
+    @staticmethod
+    def nvme_devs_count():
+        """
+        Returns count of nvme devices in the system
+        """
+        cmd = "nvme list"
+        output = process.system_output(cmd, shell=True, ignore_status=True)
+        count = len(output.splitlines()) - 2
+        print "Count: %d" % count
```

Comment:
can this be in `self.log.info `

Reply:
@PraveenPenguin Thanks for catching this. I need to remove this.